mountpoint-s3/src/prefetch.rs

```diff
@@ -535,6 +459,77 @@ mod tests {
         max_forward_seek_distance: u64,
         #[proptest(strategy = "1u64..4*1024*1024")]
         max_backward_seek_distance: u64,
+        #[proptest(strategy = "data_cache_config_strategy()")]
+        data_cache_config: DataCacheConfig,
+    }
+
+    fn data_cache_config_strategy() -> impl Strategy<Value = DataCacheConfig> {
+        prop_oneof![
+            Just(DataCacheConfig::NoCache),
+            (16usize..2 * 1024 * 1024).prop_map(|block_size| DataCacheConfig::InMemoryCache { block_size })
+        ]
+    }
+
+    type GetObjectFn<E> = dyn Fn(&str, &str, u64, ETag) -> Box<dyn PrefetchGetObject<ClientError = E>>;
+
+    struct PrefetcherBox<E> {
+        get_fn: Box<GetObjectFn<E>>,
+    }
+
+    impl<E> PrefetcherBox<E>
+    where
+        E: std::error::Error + Send + Sync + 'static,
+    {
+        fn new<Feed>(part_feed: Feed, config: PrefetcherConfig) -> Self
+        where
+            Feed: ObjectPartFeed<ClientError = E> + Send + Sync + 'static,
+        {
+            let prefetcher = Prefetcher::new(part_feed, config);
+            PrefetcherBox {
+                get_fn: Box::new(move |bucket: &str, key: &str, size: u64, etag: ETag| {
+                    Box::new(prefetcher.get(bucket, key, size, etag))
+                }),
+            }
+        }
+
+        fn get(&self, bucket: &str, key: &str, size: u64, etag: ETag) -> Box<dyn PrefetchGetObject<ClientError = E>> {
+            (self.get_fn)(bucket, key, size, etag)
+        }
+    }
```

Comment:
not sure what the purpose of this is, it could use some rustdoc

Reply:
You're right. This is a wrapper I quickly built to dynamically switch between prefetchers with and without cache in the tests. I'll either rethink the tests or document it.